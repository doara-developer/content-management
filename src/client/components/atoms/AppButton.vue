<template>
    <button class="app-button" @click="click"><slot /></button>
</template>
<script lang="ts">
import Vue from "vue";

export default Vue.extend({
    methods: {
        click() {
            this.$emit("click");
        },
    },
});
</script>
