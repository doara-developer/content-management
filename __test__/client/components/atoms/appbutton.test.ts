import { shallowMount } from "@vue/test-utils";
import AppButton from "@client/components/atoms/AppButton.vue";

describe("AppButton.vue", () => {
    test("renders props.msg when passed", () => {
        const wrapper = shallowMount(AppButton);
        expect(wrapper).toMatchSnapshot();
    });
});
