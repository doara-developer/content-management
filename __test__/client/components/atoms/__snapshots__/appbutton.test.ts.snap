// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton.vue renders props.msg when passed 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
